app/src/main/java/org/mozilla/fenix/settings/search/RadioSearchEngineListPreference.kt

```diff
@@ -99,12 +103,18 @@ class RadioSearchEngineListPreference @JvmOverloads constructor(
         isSelected: Boolean,
     ): View {
         val isCustomSearchEngine = engine.type == SearchEngine.Type.CUSTOM
+        val isTopicSpecific = !(engine.isGeneral || isCustomSearchEngine)
```

Comment:
The `isGeneral` check should already account for the CUSTOM type of a search engine - https://github.com/mozilla-mobile/firefox-android/blob/f99a255860a9faa4a9845f206a6a7b4694871bc6/android-components/components/feature/search/src/main/java/mozilla/components/feature/search/storage/SearchEngineReader.kt#L74-L78.

I understand the `isGeneral` property may not easily transmit this. We could have a new extension property for a search engine - `isTopicSpecific` that would just be `!isGeneral` if this will make the functionality easier to reason about.